src/couch_views/src/couch_views.erl

```diff
@@ -0,0 +1,115 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_views).
+
+-export([
+    map_query/6
+]).
+
+-include("couch_views.hrl").
+
+
+map_query(Db, DDoc, ViewName, Callback, Acc0, Args0) ->
+    Args = process_args(Args0),
+    #{name := DbName} = Db,
+    {ok, Mrst} = couch_views_util:ddoc_to_mrst(DbName, DDoc),
```

Comment:
we call this later in

`couch_views_worker`:
```
get_indexing_info(JobData) ->
    #{
        <<"db_name">> := DbName,
        <<"ddoc_id">> := DDocId
    } = JobData,
    {ok, Db} = fabric2_db:open(DbName, []),
    {ok, DDoc} = fabric2_db:open_doc(Db, DDocId),
    {ok, Mrst} = couch_views_util:ddoc_to_mrst(DbName, DDoc),
    {ok, Db, Mrst}.
```

When you create the JobData, you already have access to Mrst:

```
create_job_data(Db, Mrst, LastSeq) ->
    #{name := DbName} = Db,

    #mrst{
        idx_name = DDocId
    } = Mrst,

    #{
        db_name => DbName,
        ddoc_id => DDocId,
        last_seq => LastSeq
    }.
```
Curious why you don't just pass it in?

Reply:
Are you asking why not store the `Mrst` in fdb for the worker?